httpie/utils.py

```diff
@@ -12,10 +12,29 @@
 RE_COOKIE_SPLIT = re.compile(r', (?=[^ ;]+=)')
 
 
+class JsonDictWithDuplicateKeys(OrderedDict):
+    """A specialized JSON dict with duplicate keys support.
+
+    """
+
+    def __init__(self, items):
+        # `json.dumps()` will check the underlying dict (`self` in our case) and outputs
+        # directly if it is empty. So we need to have something in the dictionary.
+        self[''] = ''
```

Comment:
Is there a bool check somewhere? I think something like this with a docstring would be cleaner — as far as a hack of this type goes:

```python
def __bool__(self):
  return True
```

Reply:
That seems to work, thanks! :champagne:

Reply:
Finally, that is not enough. Overriding `__bool__` works only for the Python implementation of the JSON encoder. It will do a simple [`if not dct` check](https://github.com/python/cpython/blob/9d318ad3ecca69b33fb4bb23a308488ec68e82f0/Lib/json/encoder.py#L334-L336).

But most of the time, it will be the C implementation of the JSON encoder that will be used. And the check is not quite the same: it does [` if (PyDict_GET_SIZE(dct) == 0)`](https://github.com/python/cpython/blob/9d318ad3ecca69b33fb4bb23a308488ec68e82f0/Modules/_json.c#L1581-L1582), which is [defined like](https://github.com/python/cpython/blob/f8a95df84bcedebc0aa7132b3d1a4e8f000914bc/Include/cpython/dictobject.h#L48):
```c
#define PyDict_GET_SIZE(mp)  (assert(PyDict_Check(mp)),((PyDictObject *)mp)->ma_used)
```
The problem here, is that `ma_used` is updated only when an item is effectively added to the dict, and I am not aware of a workaround to fake that value.

Reply:
Is it worth adding so much technical details to the comment below `self[''] = ''`? I would say yes, WDYT?

Reply:
Right. Is there’s anything special about using specifically `''`? Maybe we could put there something more explicit like `self.['__hack__'] = '__hack__'` (also easier to debug if it ever appears in some output).

Reply:
No, I just used the smaller string available. I'll go with `__hack__` :+1: